### PR TITLE
NIFI-7815: Enhance the logging in MergeContent processor

### PR DESCRIPTION
**Pull request by pgyori**
https://issues.apache.org/jira/browse/NIFI-7815

#### Description of PR

Logging is enhanced in MergeContent processor where an exception was suppressed in case another exception was thrown while removing a flowfile from the session.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java:586]:
```suggestion
    private void removeFlowFileFromSession(final ProcessSession session, final FlowFile flowFile, final ProcessContext context) {
        try {
            session.remove(flowFile);
        } catch (final Exception e) {
            getLogger().error("Failed to remove merged FlowFile from the session after merge failure during " + context.getProperty(MERGE_FORMAT).getValue() + " merge.", e);
        }
    }
```

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java:647]:
```suggestion
                removeFlowFileFromSession(session, bundle, context);
```

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java:792]:
```suggestion
                removeFlowFileFromSession(session, bundle, context);
```

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java:859]:
```suggestion
                removeFlowFileFromSession(session, bundle, context);
```

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java:929]:
```suggestion
                removeFlowFileFromSession(session, bundle, context);
```

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeContent.java:1053]:
```suggestion
                removeFlowFileFromSession(session, bundle, context);
```

**Review by pvillard31 — COMMENTED** [review]
Instead of defining a different message for each case, can we leverage ``context.getProperty(MERGE_FORMAT).getValue()`` for the log statement? See suggestions. Feel free to ignore if not relevant.

**Comment by pgyori** [comment]
> Instead of defining a different message for each case, can we leverage `context.getProperty(MERGE_FORMAT).getValue()` for the log statement? See suggestions. Feel free to ignore if not relevant.

That's right, thank you @pvillard31 ! Committed the changes.
I added quotes around the merge format, so that when something like:
_Failed to remove merged FlowFile from the session after merge failure during "FlowFile Stream, v3" merge._
gets written in the log, it is clearer that the expression "FlowFile Stream, v3" is the name of the merge format.

**Review by pvillard31 — APPROVED** [review]
LGTM. @markap14, is it OK for you as well? Happy to merge if it LGTY.

**Review by pvillard31 — APPROVED** [review]

**Pull request closed without merging**
